src/ServiceControl.Transports.ASBS/AzureQuery.cs

```diff
@@ -291,37 +283,57 @@ async Task<IReadOnlyList<MetricValue>> GetMetrics(string queueName, DateOnly sta
             },
             cancellationToken);
 
-        var metricValues =
-            response.Value.Values.FirstOrDefault()?.Metrics.FirstOrDefault()?.TimeSeries.FirstOrDefault()?.Values ?? [];
+        var metricQueryResult = response.Value.Values.SingleOrDefault(mr => mr.Namespace == MicrosoftServicebusNamespacesMetricsNamespace);
+
+        if (metricQueryResult is null)
+        {
+            throw new Exception($"No metrics query results returned for {MicrosoftServicebusNamespacesMetricsNamespace}");
+        }
+
+        var metricResult = metricQueryResult.GetMetricByName(CompleteMessageMetricName);
+
+        if (metricResult.Error.Message is not null)
+        {
+            throw new Exception($"Metrics query result for '{metricResult.Name}' failed: {metricResult.Error.Message}");
+        }
+
+        var timeSeries = metricResult.TimeSeries.SingleOrDefault();
+
+        if (timeSeries is null)
```

Comment:
What if no messages actually been processed for that whole day? Cloudfront returns empty slots. Not sure on this api.

Maybe instead a WARN should be logged.